org.eclipse.lsp4e/src/org/eclipse/lsp4e/outline/LSSymbolsContentProvider.java

```diff
@@ -292,12 +334,45 @@ public Object[] getElements(Object inputElement) {
 		if (this.lastError != null && symbolsModel.getElements().length == 0) {
 			return new Object[] { "An error occured, see log for details" }; //$NON-NLS-1$
 		}
-		return symbolsModel.getElements();
+		return Arrays.stream(symbolsModel.getElements())
+				.filter(element -> !hideElement(element))
+				.toArray(Object[]::new);
 	}
 
 	@Override
 	public Object[] getChildren(Object parentElement) {
-		return symbolsModel.getChildren(parentElement);
+		return Arrays.stream(symbolsModel.getChildren(parentElement))
+			.filter(element -> !hideElement(element))
+			.toArray(Object[]::new);
+	}
+
+	/**
+	 * Decide whether to hide the given element or not.
+	 * Default implementation hides elements depending on a preference
+	 * that determines which {@link SymbolKind}s
+	 * (property in {@link DocumentSymbol}, {@link DocumentSymbolWithURI}s, and {@link SymbolInformation})
+	 * are to be hidden in the outline view.
+	 * Sub-classes may override this method to add language-specific filtering.
+	 *
+	 * @param element an outline view contents element to be checked
+	 * @return <code>true</code> if the given element is not to be shown in the outline view
+	 */
+	protected boolean hideElement(Object element) {
+		SymbolKind kind = null;
+
+		if (element instanceof DocumentSymbol) {
+			kind = ((DocumentSymbol) element).getKind();
+		} else if (element instanceof DocumentSymbolWithURI) {
+			kind = ((DocumentSymbolWithURI) element).symbol.getKind();
+		} else if (element instanceof SymbolInformation) {
+			kind = ((SymbolInformation) element).getKind();
+		}
+
+		if (OutlineViewHideSymbolKindMenuContributor.isHideSymbolKind(kind)) {
```

Comment:
```suggestion
		return OutlineViewHideSymbolKindMenuContributor.isHideSymbolKind(kind);
```
and remove the rest of the method.